tests/unit/testsuite/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateControllerTest.php

```diff
@@ -0,0 +1,196 @@
+            'street_address2' => 'Additional street 007',
+            'street_address3' => null,
+            'street_address4' => null,
+            'first_name'      => 'TesterFn',
+            'last_name'       => 'TesterLn',
+            'locality'        => 'Los Angeles',
+            'region'          => 'CA',
+            'postal_code'     => '10011',
+            'country_code'    => 'US',
+            'phone'           => '+1 123 111 3333',
+            'phone_number'    => '+1 123 111 3333',
+        ];
+
+        $postData = [
+            'shipping_address' => [
+                'street' => [
+                    'Test street 123',
+                    'Additional street 007',
+                ],
+                'firstname'     => 'TesterFn',
+                'lastname'      => 'TesterLn',
+                'city'          => 'Los Angeles',
+                'region_id'     => 12, // Region: CA
+                'postcode'      => '10011',
+                'country_id'    => 'US',
+                'telephone'     => '+1 123 111 3333',
+            ]
+        ];
+
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods(['getRequest'])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $result = $this->currentMock->prepareAddressData($postData);
+
+        $this->assertEquals($actual, $result);
+    }
+
+    public function testSaveActionWhereBoltReferenceIsFalse()
+    {
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods(['getRequest', 'getResponse', '_redirect'])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $request = new Mage_Core_Controller_Request_Http();
+        $request->setPost('bolt_reference', false);
+
+        $response = new Mage_Core_Controller_Response_Http();
+
+        $this->currentMock->method('getRequest')
+            ->willReturn($request);
+
+        $this->currentMock->method('getResponse')
+            ->willReturn($response);
+
+        $this->currentMock->method('_redirect')
+            ->willReturn(false);
+
+        $result = $this->currentMock->saveAction();
+
+        $this->assertNull($result);
+    }
+
+    public function testSaveAction()
+    {
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Adminhtml_Sales_Order_CreateController')
+            ->setMethods([
+                'getRequest', '_redirect', '_processActionData', '_getSession',
+                'getImmutableQuoteIdFromTransaction', '_normalizeOrderData',
+                '_getOrderCreateModel', '_postOrderCreateProcessing'
+            ])
+            ->enableOriginalConstructor()
+            ->getMock();
+
+        $session = new Mage_Adminhtml_Model_Session_Quote();
+        $request = new Mage_Core_Controller_Request_Http();
+        $request->setPost('bolt_reference', true);
```

Comment:
let's use realistic data in test - I think 'bolt_reference' is a string with form 'AAAA-BBBB-1234'

Reply:
fixed.